### DISCUSSION TITLE
aws lambda deployment and environment variables

### DISCUSSION THREAD

**Discussion by afarbman** [developmentseed/titiler#237] [Q&A] [answered]
Hi, 

Following the AWS lambda deployment guide found in here: https://devseed.com/titiler/deployment/aws/lambda/

when hitting any of the endpoints (e.g. /cog/info) I am getting a 404.
this is the response body:
{
  "detail": "'/vsis3/landsat-pds/L8/001/002/LC80010022016230LGN00/LC80010022016230LGN00_B1.TIF' not recognized as a supported file format."
}

what am I missing?

**Comment by vincentsarago** [comment, marked as answer]
@afarbman do you have `landsat-pds` in the bucket list ? 

https://github.com/developmentseed/titiler/blob/master/deployment/aws/cdk/config.py#L20

To access file via `s3://`  URL, you need to allow the lambda to do HEAD/GET operation on the bucket. see https://github.com/developmentseed/titiler-pds/blob/master/stack/config.py#L21-L26

**Reply by vincentsarago** [reply]
side note, TiTiler default configuration is not optimized for the Landsat dataset: https://github.com/developmentseed/titiler/blob/master/deployment/aws/cdk/app.py#L21-L23

landsat-pds data are not proper COG and thus we need to be able to get the external overview see: https://github.com/developmentseed/titiler-pds/blob/master/titiler_pds/routes/landsat.py#L13-L26

**Reply by afarbman** [reply]
Hi @vincentsarago ,
It turns out this was an AWS CDK misconfiguration. I did not deploy the lambda to the right VPC.
I was confused by the vsis prefix received in the response and thought there might be more to it.
anyways, deployed it to a simple ec2 and things looks good.

BTW, Awesome project!

**Reply by vincentsarago** [reply]
> I was confused by the vsis prefix received in the response

FYI: GDAL only deals with `/vsi*/` prefix not `s3://` url. When opening a file via rasterio, it will automatically transform the url from s3:// to /vsis3.

**Reply by afarbman** [reply]
In case anyone else bumps into a weird 404,
there is a settings in app.py which restricts the allowed tiff extensions:
CPL_VSIL_CURL_ALLOWED_EXTENSIONS=".tif"

**Reply by vincentsarago** [reply]
yeah, sorry about this. I'll add a note in the app.py file 

Hopefully all other config will be explained in https://github.com/developmentseed/titiler/pull/241

**Reply by ramotswa** [reply]
Sooo I have just hit this issue and I thought it was because my tiff files have a `.tiff` extension. (to `.tif` or not to `.tiff`?!). 
Either way - it doesn't seem to work - even using my favourite test case...;
```{"detail":"'/vsis3/planet-disaster-data/hurricane-harvey/SkySat_Freeport_s03_20170831T162740Z3.tif' not recognized as a supported file format."}```

I'd quite like this on Lambda rather than resorting to EC2 - so any pointers appreciated.

**Reply by vincentsarago** [reply]
did you add `planet-disaster-data` bucket in https://github.com/developmentseed/titiler/blob/master/deployment/aws/cdk/config.py#L20 ?

**Reply by ramotswa** [reply]
Ah. Lovely - all works now. Thanks.
